### PR TITLE
dev-libs/re2: add 0.2023.06.02

### PR DESCRIPTION
**Pull request by stha09**

### CODE REVIEW

File: dev-cpp/abseil-cpp/abseil-cpp-20230125.2.ebuild [hunk 1]
```diff
@@ -56,13 +56,17 @@ src_prepare() {
...
+			-DABSL_ENABLE_INSTALL=TRUE
+			-DABSL_LOCAL_GOOGLETEST_DIR="${WORKDIR}/googletest-${GTEST_COMMIT#v}"
+			-DABSL_PROPAGATE_CXX_STD=TRUE
+			-DABSL_BUILD_TESTING=$(usex test ON OFF)
+			$(usex test -DBUILD_TESTING=ON '') #intentional usex, it used both variables for tests.
+		)
+		cmake_src_configure
+	}
+
+	multilib_parallel_foreach_abi abi_configure
```
Comment (thesamesam) [inline dev-cpp/abseil-cpp/abseil-cpp-20230125.2.ebuild:71]: `multilib_parallel_foreach_abi` is deprecated, but could you try defining `multilib_src_configure` instead, then we don't need the nesting? (hoist abi_configure into top-level, rename to multilib_src_configure).

File: dev-qt/qtwebengine/qtwebengine-5.15.9_p20230505.ebuild [hunk 2]
```diff
@@ -42,7 +42,7 @@ RDEPEND="
 	dev-libs/libevent:=
 	dev-libs/libxml2[icu]
 	dev-libs/libxslt
-	dev-libs/re2:=
+	<dev-libs/re2-2023-06-02:=
```
Comment (thesamesam) [inline dev-qt/qtwebengine/qtwebengine-5.15.9_p20230505.ebuild:45]: Could you add a comment noting what you said on IRC wrt the C++ standard versions?

**Review by thesamesam — COMMENTED** [review]
Thanks for looking at this!

**Comment by stha09** [comment]
I need to double check if it helps to compile re2 in C++14 mode. With cmake re2 is compiled with gnu++14, but with make it uses compiler default.

**Comment by stha09** [comment]
> I need to double check if it helps to compile re2 in C++14 mode. With cmake re2 is compiled with gnu++14, but with make it uses compiler default.

Doesn't help, because chromium is compiled in C++20 mode.

**Comment by Pesa** [comment]
Is something forcing c++14 for qtwebengine? The rest of Qt5 should be using the highest standard version supported by the current compiler (up to c++17 or 20, don't remember), but qtwebengine is a bit special...
/cc @Chiitoo

**Comment by stha09** [comment]
> Is something forcing c++14 for qtwebengine? The rest of Qt5 should be using the highest standard version supported by the current compiler (up to c++17 or 20, don't remember), but qtwebengine is a bit special... /cc @Chiitoo

I haven't tried, but Chromium enabled C++17 beginning 2022 (around M98-M100). qtwebengine is based on M87. I give it a try.

**Comment by ionenwks** [comment]
> Is something forcing c++14 for qtwebengine? The rest of Qt5 should be using the highest standard version supported by the current compiler (up to c++17 or 20, don't remember), but qtwebengine is a bit special...

*If needed*, on IRC we were talking about how it wouldn't be that bad to use bundled re2 on old slot 5 -- much like we ended up doing for ffmpeg given there's not really anyone to update old code for new libraries unlike slot 6.

Not great but it'll be hard to keep around for any extended period (before eventual removal) if issues like that keep coming up.

**Comment by stha09** [comment]
I tried to compile with C++17. It works with few modifications. I only don't get it managed that the qmake generated makefiles use C++17. Those always use C++1y (C++14). Someone has an idea?
On the other hand, I don't know if it really works at runtime. There are few strange warnings.

**Review by Chiitoo — APPROVED** [review]
It seems C++20 is used/should be possible to be used looking at these commits:

https://code.qt.io/cgit/qt/qtwebengine.git/commit/?h=5.15&id=08053c0a8a2cfc01c4d92155342a98d610e12473

https://code.qt.io/cgit/qt/qtwebengine.git/commit/?h=5.15&id=6369c52cebd276f03856dd333af727fd8427ac63

Don't really have anything else to add, not being familiar with re2 and all.

**Comment by a17r** [comment]
> I tried to compile with C++17. It works with few modifications. I only don't get it managed that the qmake generated makefiles use C++17. Those always use C++1y (C++14). Someone has an idea?

I haven't had time to look at it yet. Do you have a link with your modifications?

**Pull request closed without merging**